dask_mongo/core.py

```diff
@@ -9,24 +9,44 @@
 from dask.bag import Bag
 from dask.base import tokenize
 from dask.graph_manipulation import checkpoint
-
 from ._version import __version__
+from functools import lru_cache
 
 appname = f"dask-mongo/{__version__}"
 
+_CACHE_SIZE = 10
+
+
+class CachedKwargs(dict):
+    def __hash__(self):
+        return hash(frozenset(self))
+
+
+@lru_cache(_CACHE_SIZE, typed=True)
+def _cache_inner(kwargs):
+    return pymongo.MongoClient(appname=appname, **dict(kwargs))
```

Comment:
IIRC I don't think this extra `dict(...)` call is  needed 

```suggestion
    return pymongo.MongoClient(appname=appname, **kwargs)
```

Reply:
Isn't it problematic to pass recursively "frozen" kwargs to MongoClient? Any kwargs that are lists are turned into tuples and dicts are turned into frozensets. Also have you tested this with all the supported MongoClient options? I'm also wondering about AutoEncryptionOpts which itself can contain a MongoClient.